mmh, this seems to break cross-compilation for libtiff.

<details><summary>build log</summary>

```
[ 34%] Linking C shared library libtiff.so
/nix/store/xha8y6kl52riwlbn866ac7np208iysdn-aarch64-unknown-linux-gnu-binutils-2.35.1/bin/aarch64-unknown-linux-gnu-ld: skipping incompatible /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libm.so.6 when searching for /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libm.so.6
/nix/store/xha8y6kl52riwlbn866ac7np208iysdn-aarch64-unknown-linux-gnu-binutils-2.35.1/bin/aarch64-unknown-linux-gnu-ld: cannot find /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libm.so.6
/nix/store/xha8y6kl52riwlbn866ac7np208iysdn-aarch64-unknown-linux-gnu-binutils-2.35.1/bin/aarch64-unknown-linux-gnu-ld: skipping incompatible /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libm.so.6 when searching for /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libm.so.6
/nix/store/xha8y6kl52riwlbn866ac7np208iysdn-aarch64-unknown-linux-gnu-binutils-2.35.1/bin/aarch64-unknown-linux-gnu-ld: skipping incompatible /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libmvec.so.1 when searching for /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libmvec.so.1
/nix/store/xha8y6kl52riwlbn866ac7np208iysdn-aarch64-unknown-linux-gnu-binutils-2.35.1/bin/aarch64-unknown-linux-gnu-ld: cannot find /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libmvec.so.1
/nix/store/xha8y6kl52riwlbn866ac7np208iysdn-aarch64-unknown-linux-gnu-binutils-2.35.1/bin/aarch64-unknown-linux-gnu-ld: skipping incompatible /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libmvec.so.1 when searching for /nix/store/harm95yj7ag3jrr3chrwyl8kagmplssc-glibc-2.33-45/lib/libmvec.so.1
collect2: error: ld returned 1 exit status
make[2]: *** [libtiff/CMakeFiles/tiff.dir/build.make:723: libtiff/libtiff.so.5.7.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:535: libtiff/CMakeFiles/tiff.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

</details>

Not sure why that is the case though.

(note that I don't want to block on cross-compilation failures, but it's good to keep in mind).

--- a/pkgs/development/libraries/libtiff/default.nix
+++ b/pkgs/development/libraries/libtiff/default.nix
@@ -1,8 +1,8 @@
 { lib, stdenv
 , fetchurl
 
+, autoreconfHook
 , pkg-config
-, cmake
 
 , libdeflate
 , libjpeg
@@ -19,10 +19,6 @@ stdenv.mkDerivation rec {
     sha256 = "1j3snghqjbhwmnm5vz3dr1zm68dj15mgbx1wqld7vkl7n2nfaihf";
   };
 
-  cmakeFlags = if stdenv.isDarwin then [
-    "-DCMAKE_SKIP_BUILD_RPATH=OFF"
-  ] else null;
-
   # FreeImage needs this patch
   patches = [ ./headers.patch ];
 
@@ -34,7 +30,9 @@ stdenv.mkDerivation rec {
     moveToOutput include/tiffiop.h $dev_private
   '';
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  # If you want to change to a different build system, please make
+  # sure cross-compilation works first!
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
 
   propagatedBuildInputs = [ libjpeg xz zlib ]; #TODO: opengl support (bogus configure detection)
 
@@ -42,8 +40,7 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  doInstallCheck = true;
-  installCheckTarget = "test";
+  doCheck = true;
 
   meta = with lib; {
     description = "Library and utilities for working with the TIFF image file format";

--- a/pkgs/development/libraries/libtiff/default.nix
+++ b/pkgs/development/libraries/libtiff/default.nix
@@ -12,11 +12,11 @@
 
 stdenv.mkDerivation rec {
   pname = "libtiff";
-  version = "4.2.0";
+  version = "4.3.0";
 
   src = fetchurl {
     url = "https://download.osgeo.org/libtiff/tiff-${version}.tar.gz";
-    sha256 = "1jrkjv0xya9radddn8idxvs2gqzp3l2b1s8knlizmn7ad3jq817b";
+    sha256 = "1j3snghqjbhwmnm5vz3dr1zm68dj15mgbx1wqld7vkl7n2nfaihf";
   };
 
   cmakeFlags = if stdenv.isDarwin then [
@@ -47,7 +47,9 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "Library and utilities for working with the TIFF image file format";
-    homepage = "http://download.osgeo.org/libtiff";
+    homepage = "https://libtiff.gitlab.io/libtiff";
+    changelog = "https://libtiff.gitlab.io/libtiff/v${version}.html";
+    maintainers = with maintainers; [ qyliss ];
     license = licenses.libtiff;
     platforms = platforms.unix;
   };

--- a/pkgs/development/libraries/libtiff/headers.patch
+++ b/pkgs/development/libraries/libtiff/headers.patch
@@ -1,14 +1,16 @@
-diff --git i/libtiff/CMakeLists.txt w/libtiff/CMakeLists.txt
-index 90105b28..340c75cf 100755
---- i/libtiff/CMakeLists.txt
-+++ w/libtiff/CMakeLists.txt
-@@ -31,6 +31,9 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/tiffconf.h.cmake.in
-                @ONLY)
+diff --git i/libtiff/Makefile.am w/libtiff/Makefile.am
+index 44522b62..d66e5948 100644
+--- i/libtiff/Makefile.am
++++ w/libtiff/Makefile.am
+@@ -36,8 +36,11 @@ EXTRA_DIST = \
+ 	tiffconf.h.cmake.in
  
- set(tiff_public_HEADERS
-+        tiffiop.h
-+        ${CMAKE_CURRENT_BINARY_DIR}/tif_config.h
-+        tif_dir.h
-         tiff.h
-         tiffio.h
-         tiffvers.h
+ libtiffinclude_HEADERS = \
++	tif_config.h \
++	tif_dir.h \
+ 	tiff.h \
+ 	tiffio.h \
++	tiffiop.h \
+ 	tiffvers.h
+ 
+ if HAVE_CXX

--- a/pkgs/development/libraries/libtiff/headers.patch
+++ b/pkgs/development/libraries/libtiff/headers.patch
@@ -1,13 +1,14 @@
-diff -ruN a/libtiff/CMakeLists.txt b/libtiff/CMakeLists.txt
---- a/libtiff/CMakeLists.txt	2019-05-31 13:05:22.849705817 +0000
-+++ b/libtiff/CMakeLists.txt	2020-11-27 21:50:03.527831837 +0000
-@@ -42,6 +42,9 @@
-   libtiffxx.map)
+diff --git i/libtiff/CMakeLists.txt w/libtiff/CMakeLists.txt
+index 90105b28..340c75cf 100755
+--- i/libtiff/CMakeLists.txt
++++ w/libtiff/CMakeLists.txt
+@@ -31,6 +31,9 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/tiffconf.h.cmake.in
+                @ONLY)
  
- set(tiff_HEADERS
-+  tiffiop.h
-+  ${CMAKE_CURRENT_BINARY_DIR}/tif_config.h
-+  tif_dir.h
-   tiff.h
-   tiffio.h
-   tiffvers.h)
+ set(tiff_public_HEADERS
++        tiffiop.h
++        ${CMAKE_CURRENT_BINARY_DIR}/tif_config.h
++        tif_dir.h
+         tiff.h
+         tiffio.h
+         tiffvers.h

